addons/assigngear/functions/fn_loadAssignGear.sqf

```diff
@@ -0,0 +1,75 @@
+/*
+ * Name = TMF_assignGear_fnc_loadAssignGear
+ * Author = Freddo
+ *
+ * Arguments:
+ * 0: String. CfgLoadouts faction
+ * 1: String. CfgLoadouts role
+ * 2: Configfile (optional). configfile to check, can be either missionConfigFile or configFile
+ *
+ * Return:
+ * Array. Loadout array
+ *
+ * Description:
+ * Loads assigngear from config
+ */
+#include "\x\tmf\addons\assignGear\script_component.hpp"
+
+#define CFGROLE (_cfg >> "CfgLoadouts" >> _faction >> _role)
+#define CFGPARSE (configFile >> "CfgLoadoutsParser")
+
+params [
+    "_faction",
+    "_role",
+    "_cfg"
+];
+
+if (isNil "_cfg") then {
+    _cfg = if (isClass (missionConfigFile >> "CfgLoadouts" >> _faction >> _role)) then
+    [
+        {missionConfigFile},
+        {configFile}
+    ];
+};
+
+ASSERT_TRUE(isClass CFGROLE, format [ARR_3("Loadout not present: %1 %2", _faction, _role)]);
+
+private _loadout = format ["loadout_%1_%2", _faction, _role];
+private _loadoutArray = [];
+{
+    switch (toLower configName _x) do {
```

Comment:
I am not super thrilled over this code either due to pretty poor readability.
The code duplication is also quite strong in this one too, do we really need to have `_loadoutArray set [0, _x call BIS_fnc_getCfgData` listed 23 times.

Switching case to a
`private _index = switch ()...` would be better in my opinion.